core/sugartoir.ml

```diff
@@ -909,16 +911,16 @@ struct
               in
                 I.switch env (ev e, cases, t)
           | DatabaseLit (name, (None, _)) ->
-              I.database (ev (with_pos pos (RecordLit ([("name", name)],
-                                          Some (with_pos pos (FnAppl (with_pos pos (Var "getDatabaseConfig"), [])))))))
+              I.database (ev (WithPos.make ~pos (RecordLit ([("name", name)],
```

Comment:
I have experimentally replaced `with_pos` with `WithPos.make`, which is slightly more verbose. Does that look sensible? Or do we want to keep a local `with_pos` alias? Also, `WithPos.` prefix is not really mandatory here since the module is opened at the top level, but having only `make ~pos ...` looks a bit unintuitive.

Reply:
I'm happy with this change. I think having the prefix is nice, as the only real reason we open the `WithPos` namespace is to pattern match the types.